### PR TITLE
Fixing deletedSites modal not persisting while clearing sites

### LINKED ISSUE #14137: Unable to delete multiple publishers from deleted list
### Description
Issue reported by @kjozwiak . Unable to delete multiple publishers from deleted list 

### Steps to Reproduce
  1. Enable payments and add simulated payments 
  2. Delete a few publishers from the ledger table
  3. Click on show deleted publishers to bring up the deleted publishers list modal 
  4. Modal closes when you delete a single publisher from the list

**Actual result:**
issue

**Expected result:**
Add a checkbox where user can choose multiple publisher and click on delete to remove them. Something similar to the no script modal 

**Reproduces how often:**
100%

### Brave Version

**about:brave info:**

Brave | 0.22.712
-- | --
V8 | 6.6.346.26
rev | e48c5ff
Muon | 6.0.9
OS Release | 10.0.16299
Update Channel | Release
OS Architecture | x64
OS Platform | Microsoft Windows
Node.js | 7.9.0
Brave Sync | v1.4.2
libchromiumcontent | 66.0.3359.139

**Reproducible on current live release:**
Yes

### PR DESCRIPTION
**Pull request by ryanml**
Fixes #14137 

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
  1. Enable payments and add simulated payments (`npm run add-simulated-synopsis-visits 10`)
  2. Delete a few publishers from the ledger table
  3. Click on `Show Deleted Sites` to bring up the deleted publishers list modal 
  4. Delete a publisher from the list in the modal.
  5. Confirm that the modal stays visible until:
    - There are no publishers left in the list
    - The `Clear all` action is taken
    - The modal is dismissed with `Done`
  6. (Edge case test) confirm that once the deleted sites list is empty, removing a publisher from the table does not immediately trigger the modal to display.

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/renderer/components/preferences/payment/deletedSites.js [hunk 1]
```diff
@@ -26,7 +26,9 @@ class DeletedSitesContent extends ImmutableComponent {
 
   deletePermission (name, hostPattern) {
     appActions.removeSiteSetting(hostPattern, name)
-    this.props.onHide()
+    if (this.props.sites.length === 1) {
```
Comment (NejcZdovc) [inline app/renderer/components/preferences/payment/deletedSites.js]: Can you please add check if sites is defined?

Comment (ryanml) [reply app/renderer/components/preferences/payment/deletedSites.js]: A check shouldn't be needed, the sites are pulled from [this function](https://github.com/brave/browser-laptop/blob/master/app/renderer/components/preferences/paymentsTab.js#L124) and it will just stay as empty array if no deletedSites are available.

Comment (jasonrsadler) [reply app/renderer/components/preferences/payment/deletedSites.js]: From a reusability factor, is it possible another component could call into this without init'ing `sites` first?

Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/deletedSites.js]: Normally if we want that one function is used by another component, we would move it into util file. I think we are good here
